### ISSUE TITLE
False positive for use of VM managed disks in templates

### ISSUE THREAD

**Issue by BernieWhite** [microsoft/PSRule.Rules.Azure#337]
**Description of the issue**

VMs will use unmanaged disks when a vhd uri is supplied. Otherwise managed disks will be used.

The properties currently being validated, won't necessarily exist when deploying from templates. As a result a false positive is generated by `Azure.VM.UseManagedDisks`.

**Expected behaviour**

Improve conditions to validate based on a vhd uri instead.

**Module in use and version:**

- Module: PSRule.Rules.Azure
- Version: **0.10.0**

Captured output from `$PSVersionTable`:

```text
Name                           Value
----                           -----
PSVersion                      7.0.0
PSEdition                      Core
GitCommitId                    7.0.0
OS                             Microsoft Windows 10.0.18363
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```
